SimulationAgent/Simulation/DeviceStatusLogic/UpdateDeviceState.cs

```diff
@@ -108,34 +106,35 @@ private void RunInternal()
 
             try
             {
-                var scriptContext = new Dictionary<string, object>
-                {
-                    ["currentTime"] = DateTimeOffset.UtcNow.ToString("yyyy-MM-dd'T'HH:mm:sszzz"),
-                    ["deviceId"] = this.deviceId,
-                    ["deviceModel"] = this.deviceModel.Name
-                };
-
-                // until the correlating function has been called; e.g. when increasepressure is called, don't write
-                // telemetry until decreasepressure is called for that property.
-                this.log.Debug("Checking for the need to compute new telemetry",
+                // until the correlating function has been called; e.g. when increasepressure is called,
+                // don't write telemetry until decreasepressure is called for that property
+                this.log.Debug("Checking for the need to compute new sensors state",
                     () => new { this.deviceId, deviceState = actor.DeviceState });
-                if ((bool) actor.DeviceState[CALC_TELEMETRY])
+
+                if (ScriptInterpreter.IsSensorSimulationEnabled(actor.DeviceState))
```

Comment:
it seems odd that the scriptinterpreter would own whether sensor simulation is enabled.  I'd rather keep this in the actor.

Reply:
will refactor, clean up will make the PR a little bigger